ext/oj/dump.c

```diff
@@ -535,17 +535,17 @@ void oj_dump_xml_time(VALUE obj, Out out) {
     }
     if ((0 == nsec && !out->opts->sec_prec_set) || 0 == out->opts->sec_prec) {
         if (0 == tzsecs && rb_funcall2(obj, oj_utcq_id, 0, 0)) {
-            sprintf(buf,
+            int len = sprintf(buf,
```

Comment:
This change really isn't needed. I doubt the difference is enough to measure but creating a variable for something that will always be the same value has to be more expensive than a constant.

Reply:
Unfortunately, without this change, the test will fail.

```
$ rake
mkdir -p tmp/arm64-darwin21/oj/3.0.2
cd tmp/arm64-darwin21/oj/3.0.2
/Users/watson/.rbenv/versions/3.0.2/bin/ruby -I. ../../../../ext/oj/extconf.rb
>>>>> Creating Makefile for ruby version 3.0.2 on arm64-darwin21 <<<<<
checking for rb_time_timespec()... yes
checking for rb_ivar_count()... yes
checking for rb_ivar_foreach()... yes
checking for rb_gc_mark_movable()... yes
checking for stpcpy()... yes
checking for pthread_mutex_init()... yes
creating Makefile
cd -
cd tmp/arm64-darwin21/oj/3.0.2
/usr/bin/make
compiling ../../../../ext/oj/cache8.c
compiling ../../../../ext/oj/circarray.c
compiling ../../../../ext/oj/code.c
compiling ../../../../ext/oj/compat.c
compiling ../../../../ext/oj/custom.c
compiling ../../../../ext/oj/dump.c
compiling ../../../../ext/oj/dump_compat.c
compiling ../../../../ext/oj/dump_leaf.c
compiling ../../../../ext/oj/dump_object.c
compiling ../../../../ext/oj/dump_strict.c
compiling ../../../../ext/oj/err.c
compiling ../../../../ext/oj/fast.c
compiling ../../../../ext/oj/hash.c
compiling ../../../../ext/oj/hash_test.c
compiling ../../../../ext/oj/mimic_json.c
compiling ../../../../ext/oj/object.c
compiling ../../../../ext/oj/odd.c
compiling ../../../../ext/oj/oj.c
compiling ../../../../ext/oj/parse.c
compiling ../../../../ext/oj/rails.c
compiling ../../../../ext/oj/reader.c
compiling ../../../../ext/oj/resolve.c
compiling ../../../../ext/oj/rxclass.c
compiling ../../../../ext/oj/saj.c
compiling ../../../../ext/oj/scp.c
compiling ../../../../ext/oj/sparse.c
compiling ../../../../ext/oj/stream_writer.c
compiling ../../../../ext/oj/strict.c
compiling ../../../../ext/oj/string_writer.c
compiling ../../../../ext/oj/trace.c
compiling ../../../../ext/oj/util.c
compiling ../../../../ext/oj/val_stack.c
compiling ../../../../ext/oj/wab.c
linking shared-object oj/oj.bundle
cd -
mkdir -p tmp/arm64-darwin21/stage/lib/oj
install -c tmp/arm64-darwin21/oj/3.0.2/oj.bundle lib/oj/oj.bundle
cp tmp/arm64-darwin21/oj/3.0.2/oj.bundle tmp/arm64-darwin21/stage/lib/oj/oj.bundle

##########################################################################################
ruby test/tests.rb && ruby test/tests_mimic.rb && ruby test/tests_mimic_addition.rb
Run options: --seed 31807

# Running:

.......................................................................................F................................................................................................................................................................................................................................................................................................................................................................................................................

Finished in 0.026347s, 18522.0329 runs/s, 34842.6766 assertions/s.

  1) Failure:
Juice#test_time_years [/Users/watson/prj/oj/test/test_various.rb:407]:
Expected: "-2020-03-01T15:14:13Z"
  Actual: "-2020-03-01T15:14:13"

488 runs, 918 assertions, 1 failures, 0 errors, 0 skips

....
```

Before, I think the wong length was passed into ob_dump_str().

Reply:
That is an interesting observation. RFC3339 doesn't support negative years although it does make sense to so I'd have to agree.